utils/unittest/unittest.go

```diff
@@ -394,19 +394,38 @@ func NetworkTopology() network.Topology {
 
 // CrashTest safely tests functions that crash (as the expected behavior) by checking that running the function creates an error and
 // an expected error message.
-func CrashTest(t *testing.T, scenario func(*testing.T), expectedErrorMsg string, testName string) {
+func CrashTest(
+	t *testing.T,
+	scenario func(*testing.T),
+	expectedErrorMsg string,
+	expectedStatus ...int,
+) {
 	if os.Getenv("CRASH_TEST") == "1" {
 		scenario(t)
 		return
 	}
 
-	cmd := exec.Command(os.Args[0], "-test.run="+testName)
+	cmd := exec.Command(os.Args[0], "-test.run="+t.Name())
 	cmd.Env = append(os.Environ(), "CRASH_TEST=1")
 
 	outBytes, err := cmd.Output()
 	// expect error from run
 	require.Error(t, err)
-	require.Contains(t, "exit status 1", err.Error())
+
+	if len(expectedStatus) == 0 {
+		// any non-zero status is expected
+		require.Greater(t, cmd.ProcessState.ExitCode(), 0)
```

Comment:
NotEqual rather than Greater ?

Reply:
Good point, given that it is `int` it can in theory be negative.